### DISCUSSION TITLE
Extend ItemStackImpl and Builder

### DISCUSSION THREAD

**Discussion by DasLixou** [Minestom/Minestom#1052] [Help] [answered]
Hello! I want to add a class that extends ItemStackImpl for my utils, but it doesnt work cause its package-private. How can i fix this?

Thanks :D

**Comment by TheMode** [comment, marked as answer]
This is technically not possible due to the protocol.

You can use the Tag API to apply custom data to items.

**Reply by DasLixou** [reply]
But whats the problem with the protocol? i just want a list of functions that executes when clicking on the matching item. How else could i implement this?

**Reply by TheMode** [reply]
You put an unique tag on the item, and retrieve it when clicked.
